### PR TITLE
mgr/cephadm: fix prometheus haproxy/ingress deps update

### PR DESCRIPTION
**Pull request by rkachach**
Fixes: https://tracker.ceph.com/issues/58001

Signed-off-by: Redouane Kachach <rkachach@redhat.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [x] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by rkachach** [comment]
jenkins test make check

File: src/pybind/mgr/cephadm/services/monitoring.py [hunk 1]
```diff
@@ -409,8 +409,10 @@ def calculate_deps(self) -> List[str]:
         # add an explicit dependency on the active manager. This will force to
         # re-deploy prometheus if the mgr has changed (due to a fail-over i.e).
         deps.append(self.mgr.get_active_mgr().name())
-        deps += [s for s in ['node-exporter', 'alertmanager', 'ingress']
+        deps += [s for s in ['node-exporter', 'alertmanager']
                  if self.mgr.cache.get_daemons_by_service(s)]
+        if len(self.mgr.cache.get_daemons_by_type('ingress')) > 0:
+            deps.append('ingress')
```
Comment (mgfritch) [inline src/pybind/mgr/cephadm/services/monitoring.py:415]: ah subtle! the func names are unfortunate here:

- `get_daemons_by_service` is really `get_daemons_by_service_name`
- `get_daemons_by_type` is really `get_daemons_by_service_type` 

... the last one really seems to imply a param of a daemon type, when that's not actually the case.

**Review by mgfritch — APPROVED** [review]

**Review by adk3798 — APPROVED** [review]

File: src/pybind/mgr/cephadm/services/monitoring.py (hunk 1 above)
Comment (rkachach) [reply src/pybind/mgr/cephadm/services/monitoring.py:415]: > 

upsss, good catch! I didn't know about that, I think we should definitely rename them as indeed they are very confusing

**Comment by rkachach** [comment]
jenkins test api

**Comment by rkachach** [comment]
jenkins test api

**Comment by rkachach** [comment]
jenkins test dashboard cephadm

**Comment by adk3798** [comment]
https://pulpito.ceph.com/adking-2023-01-21_05:38:21-orch:cephadm-wip-adk-testing-2023-01-20-1359-distro-default-smithi/

Lots of failures (13) but all accounted for

- 3 instances of `'list' object has no attribute 'get'`. Failure was on new test for setting mon crush location introduced by a PR in the run. That PR can't be merged, but this shouldn't block merging anything else.
- 2 instances of test_cephadm.sh test failing, caused by the same mon crush location PR. Failed with `local variable 'config_json' referenced before assignment`
- 1 instance of https://tracker.ceph.com/issues/49287, known issue
- 2 instances of https://tracker.ceph.com/issues/58524, known issue
- 2 instances of https://tracker.ceph.com/issues/58535, known issue
- 1 instance of https://tracker.ceph.com/issues/57771, known issue
- 1 instance of https://tracker.ceph.com/issues/57755, known issue
- 1 failed basic smoke test that timed out waiting for all 8 OSDs to come up. Did 5 reruns of this test with the same build (https://pulpito.ceph.com/adking-2023-01-22_15:28:20-orch:cephadm-wip-adk-testing-2023-01-20-1359-distro-default-smithi/) all of which passed. My guess is an I/O error creating the last OSD, which I've seen happen (infrequently) when we make a bunch of OSDs on the split up nvme drive we use for having multiple OSDs on these machines. Considering it a 1 off for now, and not blocking merging over it.

Overall, PRs in the run should be okay to merge other than the mon crush location one causing failures. Will start to try and clean up the test suite now that we're able to make builds and run tests again.

**Comment by adk3798** [comment]
jenkins retest this please

**Pull request merged**